src/test/unit/rx_spi_expresslrs_unittest.cc

```diff
@@ -0,0 +1,463 @@
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <limits.h>
+
+extern "C" {
+    #include "platform.h"
+
+    #include "drivers/io.h"
+    #include "common/filter.h"
+
+    #include "pg/pg.h"
+    #include "pg/pg_ids.h"
+    #include "pg/rx_spi.h"
+    #include "pg/rx_spi_expresslrs.h"
+
+    #include "rx/rx_spi.h"
+    #include "rx/expresslrs.h"
+    #include "rx/expresslrs_impl.h"
+
+    #include "drivers/rx/rx_sx127x.h"
+    #include "drivers/rx/rx_sx1280.h"
+
+    extern uint8_t FHSSsequence[ELRS_NR_SEQUENCE_ENTRIES];
+    extern uint16_t crc14tab[ELRS_CRC_LEN];
+
+    extern elrsReceiver_t receiver;
+    static const elrsReceiver_t empty = elrsReceiver_t();
+
+    static rxRuntimeState_t config = rxRuntimeState_t();
+    static rxSpiExtiConfig_t extiConfig;
+    static const rxSpiConfig_t injectedConfig = {
+        .extiIoTag = IO_TAG(PA0),
+    };
+}
+
+#include "unittest_macros.h"
+#include "gtest/gtest.h"
+
+//make clean test_rx_spi_expresslrs_unittest
+TEST(RxSpiExpressLrsUnitTest, TestCrc14)
+{
+    uint16_t expectedCrc14Tab[ELRS_CRC_LEN] = {
+        0,28247,62201,40110,52133,42482,14684,22283,
+        38730,63773,26035,3044,23791,12984,44566,49217,
+        11924,16579,56429,45626,58673,35686,6088,31135,
+        47582,55177,19239,9584,29307,7212,32898,61141,
+        29567,7464,33158,61393,47322,54925,18979,9332,
+        58421,35426,5836,30875,12176,16839,56681,45886,
+        24043,13244,44818,49477,38478,63513,25783,2784,
+        51873,42230,14424,22031,260,28499,62461,40362,
+        51369,42750,14928,21511,780,27995,61941,40866,
+        24547,12724,44314,49997,37958,64017,26303,2280,
+        58941,34922,5316,31379,11672,17359,57185,45366,
+        29047,7968,33678,60889,47826,54405,18475,9852,
+        48086,54657,18735,10104,28787,7716,33418,60637,
+        11420,17099,56933,45106,59193,35182,5568,31639,
+        38210,64277,26555,2540,24295,12464,44062,49737,
+        520,27743,61681,40614,51629,43002,15188,21763,
+        37202,65285,25515,3580,23287,13472,43022,50777,
+        1560,26703,62689,39606,52669,41962,16196,20755,
+        49094,53649,19775,9064,29795,6708,34458,59597,
+        10380,18139,55925,46114,58153,36222,4560,32647,
+        57901,35962,4308,32387,10632,18399,56177,46374,
+        30055,6960,34718,59849,48834,53397,19515,8812,
+        52409,41710,15936,20503,1820,26955,62949,39858,
+        23539,13732,43274,51037,36950,65025,25263,3320,
+        23035,14252,43778,50517,37470,64521,24743,3824,
+        52913,41190,15432,21023,1300,27459,63469,39354,
+        30575,6456,34198,60353,48330,53917,20019,8292,
+        57381,36466,4828,31883,11136,17879,55673,46894,
+        10884,17619,55421,46634,57633,36726,5080,32143,
+        48590,54169,20279,8544,30315,6204,33938,60101,
+        1040,27207,63209,39102,53173,41442,15692,21275,
+        37722,64781,24995,4084,22783,13992,43526,50257
+    };
+
+    generateCrc14Table();
+    for(int i=0; i < ELRS_CRC_LEN; i++) {
```

Comment:
should have a space between for and ( and spaces around =

Reply:
fixed